### ISSUE TITLE
Change license to LGPL

### ISSUE THREAD

**Issue by blag** [mitchellrj/python-pgp#12]
Hi, I'd like to use and distribute this in a product without open sourcing my entire project. I cannot legally do that since this project is licensed under the GPL 3.0, so unfortunately I won't be using it under its current license.

If we relicensed it under the LGPL 3.0 I would be able to use it without open sourcing my entire project. That license would also require <sup>1</sup> me to submit any changes I make to python-pgp itself back upstream, which I'm fine with. <sup>2</sup>

To change the license, anybody who has ever contributed code needs to agree to the changes. Luckily, as far as I can tell, that's really only the two official contributors:

* @mitchellrj 
* @wdoekes 

If both of you are fine with changing the license to the LGPL (or something else if you have a better license in mind), please drop a confirmation/rejection comment here.

And if either of you know of anybody else who contributed code please at-mention them here as well.

Thanks! 😄 

<sup>1</sup> There are other options, but submitting my changes back upstream would be the easiest and most preferable one for me, and (I think) the most preferable to you as well.

<sup>2</sup> I'd do this anyway even if I wasn't legally obligated to do so by the license. Feel free to check out my GitHub profile - I'm a GitHub/PyPI contributor for a number of more liberally licensed projects.

**Comment by wdoekes** [comment]
I don't mind. LGPL is fine with me.

**Comment by blag** [comment]
@mitchellrj: What do you think of this licensing change?

Also, I'm happy to help maintain this project if you need any help.

**Comment by mitchellrj** [comment]
I have no problem with this in principle but I do need to read into it a bit more. Keep pestering me about it :)

**Comment by blag** [comment]
Hey @mitchellrj, have you read up on it a bit more? I'm happy to answer any questions you might have.

The only differences between the GPL and LGPL licenses reduce down to a philosophical difference: 

|       | Users | Advantage | Disadvantage |
| --- | ----- | --------- | ------------ |
| **GPL** | all users open their code | furthering RMS' goals of ubiquitous GPL software | limit your users to those who *can* license their code under the GPL |
| **LGPL** | users don't have to open their code | doesn't further RMS' goals | attract more users - those who can't or don't want to GPL license their code |

Sorry to pester you about it, but I'd like to use this package to allow users to upload their PGP public keys so I can encrypt any email my project sends them (especially password reset emails and notifications of sensitive changes). I've been working on other parts of my project to give you more time to think, but I'm quickly finishing up everything else I want to do.

All I'm really trying to do is to integrate python-pgp with [django-ogmios](https://github.com/fusionbox/django-ogmios) to sign and encrypt registration emails, password resets, and notifications about changes to sensitive settings. The [django-email-extras](https://github.com/stephenmcd/django-email-extras) and [django-gnupg-mails](https://github.com/jandd/django-gnupg-mails) projects would work perfectly for my tasks, but both of them rely on [python-gnupg](https://github.com/isislovecruft/python-gnupg) which is licensed under the GPLv3.

At the end of the day, the final decision is about which is more important to you: try to force the spread the GPL, or enable as many end users as possible to use modern encryption even if they are using closed source projects.

Thanks! 😄

**Comment by blag** [comment]
I just figured out that django-email-extras and django-gnupg-mails both rely on the PyPI version of python-gnupg, which is BSD licensed.

Isis' fork (which I linked to in my last comment) is licensed under the GPLv3, but it's not the fork that is registered to PyPI.

I no longer need to use this project or change it's license. Sorry for the noise!

@wdoekes - Thanks for being onboard with potentially changing the license for me. I appreciate it.
@mitchellrj - Thanks for considering changing the license. I know how difficult of a decision that can be.
